### PR TITLE
debugging: add note that both CW plugins consume retries on create gr…

### PR DESCRIPTION
**Pull request by PettitWesley**
…oup/stream failures

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Claych — APPROVED** [review]

File: troubleshooting/debugging.md [hunk 1]
```diff
@@ -630,6 +631,18 @@ Path                /logs/request.log
 
 In other cases, the original config would be correct. For example, assume your app produces first a log file `request.log` and then next creates and writes new logs to a new file named `request.log1` and then next `request.log2`. And then when the max number of files are reached, the original `request.log` is deleted and then a new `request.log` is written. If that is the scenario, then the original Path pattern with a wildcard at the end is correct since it is necessary to match all log files where new logs are written. This method of log rotation is the behavior of the [log4j DirectWriteRolloverStrategy](https://logging.apache.org/log4j/2.x/manual/appenders.html#DirectWriteRolloverStrategy) where there is no file renaming. 
 
+#### Both CloudWatch Plugins: Create failures consume retries
+
+In Fluent Bit, [retries are handled by the core engine](https://docs.fluentbit.io/manual/administration/scheduling-and-retries). Whenever the plugin encounters any error, it returns a retry to the engine which schedules a retry. This means that log group creation, log stream creation or log retention policy calls can consume a retry if they fail.
```
Comment (ShelbyZ) [inline troubleshooting/debugging.md:636]: Do these calls only happen on startup? Would they only effect the first batch of logs sent?

File: troubleshooting/debugging.md [hunk 2]
```diff
@@ -630,6 +631,18 @@ Path                /logs/request.log
...
+
+*This applies to both the [CloudWatch Go Plugin and the CloudWatch C Plugin](#aws-go-plugins-vs-aws-core-c-plugins).*
+
+This means that before the plugin even tries to send a chunk of log records, it may consume a retry on a random temporary networking error trying to call CreateLogStream.
```
Comment (ShelbyZ) [inline troubleshooting/debugging.md:640]: I don't know if it would feel clearer to provide an example scenario with default or set number of retries.

**Review by ShelbyZ — APPROVED** [review]

File: troubleshooting/debugging.md (hunk 1 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:636]: uhhh good point I will add a note. Depends on your config. With templating you can get lots of new groups and streams.

File: troubleshooting/debugging.md (hunk 2 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:640]: I think that just makes it too long

**Pull request merged**